### PR TITLE
Follow-up nits from etcd init script pull request

### PR DESCRIPTION
**Pull request by JamesLaverack**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

This is a follow on from the review of https://github.com/cilium/cilium/pull/29109

### CODE REVIEW

File: clustermesh-apiserver/etcdinit/root.go [hunk 1]
```diff
@@ -154,14 +154,18 @@ func InitEtcdLocal() (returnErr error) {
...
 		log.Info("Sending SIGTERM signal to etcd process")
 		err := etcdCmd.Process.Signal(syscall.SIGTERM)
 		if err != nil {
 			log.WithError(err).
 				Error("Failed to send SIGTERM signal to etcd process")
-			returnErr = err
+			// If we're not already returning an error, return an error here
+			if returnErr == nil {
+				returnErr = err
+			}
```
Comment (gandro) [inline clustermesh-apiserver/etcdinit/root.go]: Could we also use `errors.Join` here?

**Review by gandro — APPROVED** [review]
Helm LGTM! Thanks!

File: clustermesh-apiserver/etcdinit/root.go (hunk 1 above)
Comment (JamesLaverack) [reply clustermesh-apiserver/etcdinit/root.go]: I always forget about that one. If `returnErr` is `nil`, does `errors.Join` handle that case correctly?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Changes look OK, but can you please motivate them in the commit description? Going all over the comments in #29109 to understand what's going on here doesn't make for a good review experience (or for understanding the change in the future, when looking at Git logs). :pray:

**Review by aanm — APPROVED** [review]

File: clustermesh-apiserver/etcdinit/root.go (hunk 1 above)
Comment (JamesLaverack) [reply clustermesh-apiserver/etcdinit/root.go]: I checked, and it does handle that case. I've updated all instances of it in that deferred function.

**Comment by JamesLaverack** [comment]
> Changes look OK, but can you please motivate them in the commit description? Going all over the comments in #29109 to understand what's going on here doesn't make for a good review experience (or for understanding the change in the future, when looking at Git logs). 🙏

Good point. I've updated the commit message. It now reads:

```
    Follow-up nits from etcd init script pull request

    This is a follow on from the review of
    https://github.com/cilium/cilium/pull/29109, and is a collection of
    minor changes:
    - Remove unused variables in install/kubernetes Makefile values file
    - Remove etcd image from check-docker-images.sh script
    - Remove now-unused external dependencies block from
      check-docker-images.sh script
    - Clarify doc comment for ClusterMeshEtcdInit function, to correctly
      state the purpose of the hasConfig key
    - defer etcdClient.Close() after creating etcdClient
    - Use errors.Join to handle errors in defered cleanup code, where the
      main function may have also returned an error
    - Correct typo in comment: "it's" to "its"

    Signed-off-by: James Laverack <james@isovalent.com>
```

**Review by youngnick — APPROVED** [review]

**Review by marseel — APPROVED** [review]
Looks good to me.

cc @giorio94 as it addresses his comments in other PR.

**Review by giorio94 — APPROVED** [review]
Thanks!

**Review by qmonnet — APPROVED** [review]
Thanks a lot!

**Comment by giorio94** [comment]
Conformance K8s Upstream Network is https://github.com/cilium/cilium/issues/29523 (disabled by https://github.com/cilium/cilium/pull/29524)

**Comment by giorio94** [comment]
One of the Conformance E2E tests seemed to consistently fail due to https://github.com/cilium/cilium/issues/27762, although it cannot be related to this PR (as the changes here are not exercised by that test). Let's try rebasing onto main.

**Pull request merged**